expression/builtin_miscellaneous.go

```diff
@@ -525,7 +525,39 @@ type builtinIsIPv4MappedSig struct {
 // eval evals a builtinIsIPv4MappedSig.
 // See https://dev.mysql.com/doc/refman/5.7/en/miscellaneous-functions.html#function_is-ipv4-mapped
 func (b *builtinIsIPv4MappedSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("IS_IPV4_MAPPED")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return types.Datum{}, errors.Trace(err)
+	}
+
+	arg := args[0]
+	d.SetInt64(0)
+	if arg.IsNull() {
+		return d, nil
+	}
+
+	ipAddress, err := arg.ToBytes()
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+
+	if len(ipAddress) != net.IPv6len {
```

Comment:
mysql> select IS_IPV4_MAPPED("127.0.0.1");
+-----------------------------+
| IS_IPV4_MAPPED("127.0.0.1") |
+-----------------------------+
|                           0 |
+-----------------------------+
1 row in set (0.04 sec)

mysql> 

MySQL returns 0 for IPv4 address, not null.